app/api/v2/handlers/fact_api.py

```diff
@@ -86,10 +86,11 @@ async def add_facts(self, request: web.Request):
             if 'source' not in fact_data:
                 new_fact.source = WILDCARD_STRING
             new_fact.origin_type = OriginType.USER
+            await self._api_manager.verify_operation_state(new_fact)
             await knowledge_svc_handle.add_fact(new_fact)
             store = await knowledge_svc_handle.get_facts(criteria=dict(trait=new_fact.trait,
                                                                        value=new_fact.value,
-                                                                       source=WILDCARD_STRING,
+                                                                       source=new_fact.source,
```

Comment:
I think this is going to result in a minor change in behavior, but I think change you'd made is more correct than the previous option, so I'm fine with it.